src/main/java/com/google/sps/model/TasksClientImpl.java

```diff
@@ -22,53 +22,66 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public final class TasksUtility {
-  private TasksUtility() {}
+public class TasksClientImpl implements TasksClient {
 
-  /**
-   * Get instance of Tasks Service
-   *
-   * @param credential valid Google credential object with user's accessKey inside
-   * @return Google Tasks service instance
-   */
-  public static Tasks getTasksService(Credential credential) {
+  private final Tasks tasksService;
+
+  private TasksClientImpl(Credential credential) {
     HttpTransport transport = UrlFetchTransport.getDefaultInstance();
     JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
     String applicationName = ServletUtility.APPLICATION_NAME;
 
-    return new Tasks.Builder(transport, jsonFactory, credential)
-        .setApplicationName(applicationName)
-        .build();
+    tasksService =
+        new Tasks.Builder(transport, jsonFactory, credential)
+            .setApplicationName(applicationName)
+            .build();
   }
 
   /**
-   * Get a list of all TaskLists in the user's account
+   * Get all tasks from a tasklist in a user's Tasks account.
    *
-   * @param tasksService a valid Google Tasks service instance
-   * @return a list of TaskLists that belong to the user's account
-   * @throws IOException if an issue occurs with the Tasks service
+   * @param taskList TaskList object that contains the desired tasks
+   * @return List of all tasks in the tasklist
+   * @throws IOException if an issue occurs with the TasksService
    */
-  public static List<TaskList> listTaskLists(Tasks tasksService) throws IOException {
-    List<TaskList> taskLists = tasksService.tasklists().list().execute().getItems();
+  @Override
+  public List<Task> listTasks(TaskList taskList) throws IOException {
+    // returns null if no tasks exist. Convert to empty list for ease.
+    List<Task> tasks = tasksService.tasks().list(taskList.getId()).execute().getItems();
 
-    return taskLists != null ? taskLists : new ArrayList<>();
+    return tasks != null ? tasks : new ArrayList<>();
   }
 
   /**
-   * Get a list of all tasks in a specific TaskList from a user's account
+   * Get all tasklists in a user's Tasks account.
    *
-   * @param tasksService a valid Google Tasks service instance
-   * @param taskList a TaskList object that represents a user's tasklist
-   * @return a list of Tasks that belong to the specified list (empty list if no tasks present)
-   * @throws IOException if an issue occurs with the Tasks service
+   * @return List of all tasklists
+   * @throws IOException if an issue occurs with the TasksService
    */
-  public static List<Task> listTasks(Tasks tasksService, TaskList taskList) throws IOException {
-    List<Task> tasks = tasksService.tasks().list(taskList.getId()).execute().getItems();
+  @Override
+  public List<TaskList> listTaskLists() throws IOException {
+    // returns null if no tasklists exist. Convert to empty list for ease.
```

Comment:
Might want to include null in @return javadoc comment

Reply:
Null is not returned. If a null result is passed from Google (which isn't very helpful), it is replaced with an empty arraylist (which accurately reflects the tasks in a user's account: none are present)